smarts/core/sensors.py

```diff
@@ -127,6 +127,7 @@ class Vias:
 
 @dataclass
 class Observation:
+    sim_time: float
```

Comment:
Considered breaking SMARTS interface due to introducing a new field `sim_time` in class `Observation`?

Reply:
Yes, that was the intent, although I'm sorry I forgot to write about it in the PR description.  (Will edit there too, as that is significant.)

In the short term, this was useful in the `history_replacement_for_imitation_learning.py` script, however there would have been other ways to achieve the goal there without this change.  However, I still feel that it is warranted and defensible... indeed, it should have been there all along.  This way, Observations are more "portable" and can be used in more flexible ways by (possibly remote) learning agents.  For example, the can save them to a database and train with them later, perhaps filtering them based on the simulation time (for example, train on every third `Observation`), but more importantly, replay them in an external environment "in real time", that is at the clock rate they were generated at, without SMARTS.  (For example, as data inputs to a trained model that is being tested.)  For these reasons (and a few other more minor ones), I feel we want this in our Observations.

Regarding this being a "breaking" change, it's a relatively minor sort of breaking.  ;)   That is, the addition of a field to the class will not break mere users of that class who can safely ignore the additional field.  I admit that it will, however, potentially break (rigidly-typed) code that marshals these.  But the only such code of which I am aware is code that we maintain.  When I made the change I intended to go change our marshaling code too, but I forgot.  I can do so now.  However, before I push anything related to that, I'll wait for more discussion here, just in case the group feels that we should hold off on this.

Reply:
I think that the principled approach is that we do not add to the observations unless absolutely necessary until we decide to do a larger version update. If you feel this is absolutely needed we could merge this into develop and cut the release at the previous commit.

Then we would make the next release a breaking version.

Reply:
OK, since it's not absolutely necessary right now, I will work around it in `history_vehicles_replacement_for_imitation_learning.py` and back out the change to `Observations` for now.  Then I'll add an issue and a separate PR  for it so we can decide about it later.

Reply:
Added Issue #884.